[Bug]: Adding bots crashing teldrive

### Describe the bug

I have added a few bots to teldrive. Most of the time it is okay. I wanted to manage my bots manually because of rate limit and transfer limit. I don't want a ban for uploading, as the current algorithm does protect but still seem to get flagged. I noticed having too many bots actually make the performance worse. So i decided to set on 5 at a time.

When all bots added to channel seems to be fine. The bot is in good standing not banned. However when I add a certain bot this throws a panic and kill teldrive.

I checked deepwiki and it seems to be a bug.

Potential Root Causes

Looking at the code in AddBotsToChannel, the segmentation fault could be occurring in several places:

    Bot info fetching: When calling FetchBotInfo channel_manager.go:254
    Peer resolution: During peer.DefaultResolver(client.API()).ResolveDomain channel_manager.go:259
    Channel admin operations: When executing ChannelsEditAdmin channel_manager.go:288

The null pointer dereference at address 0x28 (40 in decimal) suggests accessing a struct field offset that's null, likely in the Telegram client API or peer resolution logic.
Notes

This crash appears to be a critical bug in the bot management system that needs investigation. The segmentation fault suggests improper null checking before accessing struct fields in the Telegram client interaction code. The concurrent nature of the operation (using errgroup with multiple goroutines) may also be contributing to race conditions.)

### Error Log

```
2026-04-19 08:41:10  ✓  INFO   [HTTP]   status=200 method=GET path=/api/users/profile/profile.jpeg query= ip=172.18.0.1:39722 latency=96.149257ms user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWeb...
	github.com/tgdrive/teldrive/internal/tgc/channel_manager.go:278 +0x145
2026-04-19 08:41:09  ✓  INFO   [HTTP]   status=204 method=POST path=/api/users/bots query= ip=172.18.0.1:55692 latency=1m27.33710679s user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWeb... request_size=157
2026-04-19 08:41:09  ✗  ERROR  [TG] auth.bot_failed  error=context canceled
panic: runtime error: invalid memory address or nil pointer dereference
``` 
### Version

1.8.3

### Which Platform are you using?

Linux

### Additional context

_No response_

--- a/internal/tgc/helpers.go
+++ b/internal/tgc/helpers.go
@@ -204,6 +204,15 @@ func FetchBotInfo(ctx context.Context, kvRepo repositories.KVRepository, cache c
 	if err != nil {
 		return nil, err
 	}
+	// Guard against the rare path where RunWithAuth returns nil but the
+	// inner client.Self call never actually populated `user` (for example
+	// when the connection is torn down mid-flight and the gotd middleware
+	// swallows the cancel as a no-op). Dereferencing user.ID / user.Username
+	// below was crashing the whole process because AddBotsToChannel runs
+	// the bot probe inside an errgroup that does not recover panics.
+	if user == nil {
+		return nil, fmt.Errorf("bot self lookup returned no user info")
+	}
 	return &types.BotInfo{ID: user.ID, UserName: user.Username, Token: token}, nil
 }
 

